player/command.c

```diff
@@ -216,9 +216,13 @@ static int invoke_hook_handler(struct MPContext *mpctx, struct hook_handler *h)
     char *name = mp_tprintf(22, "@%"PRIi64, h->client_id);
     int r = mp_client_send_event(mpctx, name, reply_id, MPV_EVENT_HOOK, m);
     if (r < 0) {
-        MP_WARN(mpctx, "Sending hook command failed. Removing hook.\n");
-        hook_remove(mpctx, h);
-        mp_wakeup_core(mpctx); // repeat next iteration to finish
+        if (mp_client_id_exists(mpctx, h->client_id)) {
+            MP_VERBOSE(mpctx, "Sending hook command failed. Removing hook.\n");
+            hook_remove(mpctx, h);
+        } else {
+            MP_WARN(mpctx, "Sending hook command failed.\n");
```

Comment:
We can indeed remove hooks, I think it is clients fault anyway. But could we document this behavior? That hooks are removed on any and all errors?

Also it is kinda tricky, because client doesn't have a way to know if the hooks has been removed, so it might left script in unusable state, as the hooks are not one-shoot events.

Reply:
I changed only the log level now.

Reply:
> Also it is kinda tricky, because client doesn't have a way to know if the hooks has been removed, so it might left script in unusable state,

Funnily, this is also exactly what happens when windows stops sending hooks to apps it thinks misbehave. So mpv is not the first to do that kind of thing.

I think both approaches are valid, but hooks specifically are synchronous and would definitely hurt mpv if not handled on time, so there's a bigger incentive here to break abusing clients.